File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/mutability/NestedIfStatements.java
// IGNORE_K2

import org.jetbrains.annotations.*;

class A {

    public String stringAssignment() {
        if (Math.random() > 0.50) {
            return "a string";
        }
        return "b string";
    }

    public void aVoid() {
        String aString;
        String bString = stringAssignment();
        String cString;
        if (stringAssignment().startsWith("a")) {
            aString = stringAssignment();
            cString = "aaaa";
            if (aString == bString) {
                aString = stringAssignment();
            }
        } else if (stringAssignment().startsWith("b")) {
            aString = "bbbb";
            cString = "bbbb";
        } else {
            aString = "cccc"
            cString = "cccc"
    }
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/mutability/NestedIfStatements.kt
internal class A {
    fun stringAssignment(): String {
        if (Math.random() > 0.50) {
            return "a string"
        }
        return "b string"
    }

    fun aVoid() {
        var aString: String
        val bString = stringAssignment()
        val cString: String
        if (stringAssignment().startsWith("a")) {
            aString = stringAssignment()
            cString = "aaaa"
            if (aString === bString) {
                aString = stringAssignment()
            }
        } else if (stringAssignment().startsWith("b")) {
            aString = "bbbb"
            cString = "bbbb"
        } else {
            aString = "cccc"
            cString = "cccc"
        }
    }
}
